tv_apps/tv-deck/js/tv_deck.js

```diff
@@ -297,21 +274,21 @@
       return;
     }
 
-    if (!source.onscanningstatechanged) {
-      // XXX: May change to event listener
-      source.onscanningstatechanged = function onscanningstatechanged(event) {
-        var state = event.state;
-        switch (state) {
-          case 'completed':
-          /* falls through */
-          case 'stopped':
-            this._onScanningCompleted();
-            break;
-          default:
-            break;
-        }
-      }.bind(this);
-    }
+    source.addEventListener(
+              'scanningstatechanged', function onscanningstatechanged(event) {
+      var state = event.state;
+      switch (state) {
+        case 'completed':
+        /* falls through */
+        case 'stopped':
+          this._onScanningCompleted();
+          source.removeEventListener(
+                                'scanningstatechanged', onscanningstatechanged);
```

Comment:
The function you reference here is not the one you added to event listener. Once bind() called, it returns another function. So, you cannot remove the event listener using this way.